### DISCUSSION TITLE
Error: "optional build metadata in app version must be numeric-only and cannot be greater than 65535 for msi target"

### DISCUSSION THREAD

**Discussion by hadastoroP** [tauri-apps/tauri#7600] [General]
Hello
I am trying to bundle my Tauri app and create an MSI installer for it.
I would like the version to be something like: 3.4.5+5c258740 which is a valid version by semantic versioning.
The 5c258740 represents a git commit hash.

However I get the error:
"optional build metadata in app version must be numeric-only and cannot be greater than 65535 for msi target"

any idea if I can put the hash inanother way?
Thanks

**Comment by FabianLars** [comment]
I guess that depends on how much you depend on it being unique, the range of values this hash can store is just much larger than that small number can store. So if you need to be sure it's accurate you probably just can't use it here and either think of a different versioning scheme (and maybe store the hash somewhere else in the program) or switch to the nsis setup.exe installer which supports semver versions.

If you're fine with numbers possibly overlapping (probably unlikely anyway) you could use virtually any conversation algorithm you can think of, or literally just cut that hash in half and convert it to decimal (The 65535 is the largest value you can display in hex with a length of 4)
